eclass/qt6-build.eclass

```diff
@@ -19,111 +19,143 @@ if [[ -z ${_QT6_BUILD_ECLASS} ]]; then
-		SRC_URI="https://download.qt.io/development_releases/qt/${PV%.*}/${PV/_/-}/submodules/${MY_P}.tar.xz"
-		S=${WORKDIR}/${MY_P}
+		readonly QT6_BUILD_TYPE=release
+		S=${WORKDIR}/${QT6_MODULE}-everywhere-src-${PV/_/-}
+		SRC_URI="https://download.qt.io/development_releases/qt/${PV%.*}/${PV/_/-}/submodules/${S##*/}.tar.xz"
 		;;
 	*)
 		# official stable release
-		QT6_BUILD_TYPE="release"
-		MY_P=${QT6_MODULE}-everywhere-src-${PV}
-		SRC_URI="https://download.qt.io/official_releases/qt/${PV%.*}/${PV}/submodules/${MY_P}.tar.xz"
-		S=${WORKDIR}/${MY_P}
+		readonly QT6_BUILD_TYPE=release
+		S=${WORKDIR}/${QT6_MODULE}-everywhere-src-${PV}
+		SRC_URI="https://download.qt.io/official_releases/qt/${PV%.*}/${PV}/submodules/${S##*/}.tar.xz"
 		;;
 esac
-readonly QT6_BUILD_TYPE
 
-EGIT_REPO_URI=(
-	"https://code.qt.io/qt/${QT6_MODULE}.git"
-	"https://github.com/qt/${QT6_MODULE}.git"
-)
-[[ ${QT6_BUILD_TYPE} == live ]] && inherit git-r3
-
-# @ECLASS_VARIABLE: QT6_BUILD_DIR
-# @OUTPUT_VARIABLE
-# @DESCRIPTION:
-# Build directory for out-of-source builds.
-: "${QT6_BUILD_DIR:=${S}_build}"
+if [[ ${QT6_BUILD_TYPE} == live ]]; then
+	inherit git-r3
+	EGIT_REPO_URI=(
+		"https://code.qt.io/qt/${QT6_MODULE}.git"
+		"https://github.com/qt/${QT6_MODULE}.git"
+	)
+fi
 
-IUSE="debug test"
+HOMEPAGE="https://www.qt.io/"
+LICENSE="|| ( GPL-2 GPL-3 LGPL-3 ) FDL-1.3"
+SLOT=6/$(ver_cut 1-2)
 
-if [[ ${QT6_BUILD_TYPE} == release ]]; then
-	RESTRICT="test" # bug 457182
-else
+if [[ ${PN} != qttranslations ]]; then
+	IUSE="test"
 	RESTRICT="!test? ( test )"
 fi
 
 BDEPEND="
 	dev-lang/perl
 	virtual/pkgconfig
 "
-# TODO: Tests have not been split from qtbase.
-#if [[ ${PN} != qttest ]]; then
-#	DEPEND+=" test? ( ~dev-qt/qttest-${PV} )"
-#fi
 
 ######  Phase functions  ######
 
 # @FUNCTION: qt6-build_src_prepare
 # @DESCRIPTION:
-# Prepares the environment and patches the sources if necessary.
+# Run cmake_src_prepare, prepare the environment (such as set
+# QT6_PREFIX, QT6_LIBDIR, and others), and handle anything else
+# generic as needed.
 qt6-build_src_prepare() {
-	qt6_prepare_env
-
 	cmake_src_prepare
+
+	if in_iuse test && use test && [[ -e tests/auto/CMakeLists.txt ]]; then
+		# upstream seems to install before running tests, and cmake
+		# subdir that is present in about half of the Qt6 components
+		# cause a dependency on itself and sometimes install test junk
+		sed -i '/add_subdirectory(cmake)/d' tests/auto/CMakeLists.txt || die
+	fi
+
+	_qt6_prepare_env
 }
 
 # @FUNCTION: qt6-build_src_configure
 # @DESCRIPTION:
-# Configures sources.
+# Run cmake_src_configure and handle anything else generic as needed.
 qt6-build_src_configure() {
+	if [[ ${mycmakeargs@a} == *a* ]]; then
+		local mycmakeargs=("${mycmakeargs[@]}")
+	else
+		local mycmakeargs=()
+	fi
+
+	mycmakeargs+=(
+		# note that if qtbase was built with tests, this is default ON
+		-DQT_BUILD_TESTS=$(in_iuse test && usev test ON || echo OFF)
+	)
+
+	[[ ${PN} != qttranslations ]] &&
```

Comment:
what's the story with the conditional?

Reply:
It's noted in the commit message, but the tl;dr is that it doesn't compile anything so it gives a unused cmake option warning. I could add a comment fwiw